### PR TITLE
Project submitted

### PR DESCRIPTION
**Pull request by mystycs**

### CODE REVIEW

**Comment by randallreedjr** [comment]
Hi @mystycs. There were some extra files included in your submission that make it difficult to review your code. Could you remove the `.bundle` directory and resubmit?
Alternatively, you can add me as a collaborator to your repo and I'll take care of it for you.

* Visit https://github.com/@mystycs/ttt-with-ai-project-v-000/settings/collaboration and add `randallreedjr` in the 'Collaborator' section.

Thanks!

**Comment by mystycs** [comment]
Hey there @randallreedjr  i went ahead and removed the .bundle directory.

Thanks!

File: bin/tictactoe [hunk 1]
```diff
@@ -1,3 +1,8 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+puts "Welcome to Tic Tac Toe!"
+puts "Please enter 0, 1 or 2 for number of human players."
+input = gets.chomp.to_i
+Game.new.player_count(input)
```
Comment (randallreedjr) [inline bin/tictactoe]: Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
```
Comment (randallreedjr) [inline lib/board.rb]: Another way to instantiate an Array with `i` values is `Array.new(9, " ")` (see [Array.new](http://ruby-doc.org/core-2.2.0/Array.html#method-c-new)), where `9` is the number of elements in the array and `" "` is the default value for the elements. It's more concise and will prevent errors from repeatedly typing `" "`.

File: lib/board.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
```
Comment (randallreedjr) [inline lib/board.rb]: I would just call `self.reset!` here and delegate to that method to create the default empty board.

File: lib/board.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  def turn_count
+    counter = 0
+      @cells.each do |move|
+      if (move == "X" || move == "O")
+        counter += 1
+      end
+    end
+    return counter
```
Comment (randallreedjr) [inline lib/board.rb]: Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|cell| cell == "O" || cell == "X"}
end
```

File: lib/players/human.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class Player::Human < Player
+
+  def move(board)
+    puts "Please enter your move from 1 - 9"
+    input = gets.chomp
+  end
```
Comment (randallreedjr) [inline lib/players/human.rb]: You don't need to store your return value in a local variable; just utilize the implicit return value of the expression.
```
def move(board)
  puts "Please enter your move from 1 - 9"
  gets.chomp
end
```

**Comment by randallreedjr** [comment]
@mystycs Great job! I left you a few comments. Closing this PR for now, but feel free to re-open and respond if you would like to discuss any of them.

**Pull request closed without merging**